### PR TITLE
JRU_03_01

### PR DESCRIPTION
**Pull request by Maksym82564**

### CODE REVIEW

File: src/main/java/com/tictactoe/Field.java [hunk 1]
```diff
@@ -53,7 +53,8 @@ public Sign checkWin() {
 
         for (List<Integer> winPossibility : winPossibilities) {
             if (field.get(winPossibility.get(0)) == field.get(winPossibility.get(1))
-                && field.get(winPossibility.get(0)) == field.get(winPossibility.get(2))) {
+                && field.get(winPossibility.get(0)) == field.get(winPossibility.get(2))
```
Comment (Volodymyr95) [inline src/main/java/com/tictactoe/Field.java:56]: why do u need these changes?

File: src/main/java/com/tictactoe/RestartServlet.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        logger.info("Processing POST request for /restart");
+
+        req.getSession().invalidate();
+        logger.info("Session invalidated");
+
+        resp.sendRedirect("/start");
+        logger.info("Redirected to /start");
```
Comment (Volodymyr95) [inline src/main/java/com/tictactoe/RestartServlet.java]: not needed, not informative log

File: src/main/java/com/tictactoe/TurnServlet.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+import java.io.IOException;
+import java.util.List;
+
+@WebServlet(name = "TurnServlet", value = "/turn")
+public class TurnServlet extends HttpServlet {
+    private static final Logger logger = LoggerFactory.getLogger(TurnServlet.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        logger.info("Processing GET request for /turn");
```
Comment (Volodymyr95) [inline src/main/java/com/tictactoe/TurnServlet.java]: not informative log

File: src/main/java/com/tictactoe/TurnServlet.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        logger.info("Click parameter: {}, parsed index: {}", click, index);
+        return index;
+    }
+
+    private Field extractField(HttpSession currentSession) {
+        Object fieldAttribute = currentSession.getAttribute("field");
+        if (Field.class != fieldAttribute.getClass()) {
+            currentSession.invalidate();
+            logger.error("Field attribute is invalid, invalidating session");
+            throw new RuntimeException("Field attribute is invalid");
```
Comment (Volodymyr95) [inline src/main/java/com/tictactoe/TurnServlet.java]: don't throw RuntimeException, create you own

File: src/test/java/com/tictactoe/FieldTest.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class FieldTest {
```
Comment (Volodymyr95) [inline src/test/java/com/tictactoe/FieldTest.java]: what a sense of test for Pojo? Is here a some logic?

File: src/main/java/com/tictactoe/Field.java (hunk 1 above)
Comment (Maksym82564) [reply src/main/java/com/tictactoe/Field.java:56]: To fix a bug when a sequence of 3 empty cells (Sign.EMPTY) could be mistakenly recognized as a winning sequence before the actual sequence of 3 Xs or Os.

**Review by Volodymyr95 — APPROVED** [review]

**Pull request merged**